File: app/src/main/java/com/rara/moviecatalog/MainActivity.java
package com.rara.moviecatalog;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.res.TypedArray;
import android.os.Bundle;
import android.view.View;
import android.widget.Adapter;
import android.widget.AdapterView;
import android.widget.ListView;

import com.rara.moviecatalog.adapter.MovieAdapter;
import com.rara.moviecatalog.model.Movie;

import java.util.ArrayList;

import static android.os.Looper.prepare;

public class MainActivity extends AppCompatActivity {
    private String[] dataJudul;
    private String[] dataOverview;
    private TypedArray dataPoster;
    private MovieAdapter movieAdapter;
    private ListView lvMovie;
    private ArrayList<Movie>movies;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        movieAdapter = new MovieAdapter(this);
        lvMovie = findViewById(R.id.lvMovie);

        lvMovie.setAdapter(movieAdapter);

        prepare();
        addItem();
        lvMovie.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                //Toast.makeText(MainActivity.this, movies.get(i).getJudul(),Toast.LENGTH_SHORT).show();
                Intent intent = new Intent(MainActivity.this, DetailActivity.class);
                intent.putExtra(DetailActivity.EXTRA_MOVIE, movies.get(i));
                startActivity(intent);
            }
        });

    }

    private void addItem() {
        movies = new ArrayList<>();
         for (int i=0; i < dataJudul.length; i++){
             Movie movie = new Movie();
             movie.setJudul(dataJudul[i]);
             movie.setDeskripsi(dataOverview[i]);
             movie.setPoster(dataPoster.getResourceId(i,-1));
             movies.add(movie);
         }
         movieAdapter.setMovies(movies);
    }
    private void prepare (){
        dataJudul = getResources().getStringArray(R.array.data_judul);
        dataOverview = getResources().getStringArray(R.array.data_overview);
        dataPoster = getResources().obtainTypedArray(R.array.data_poster);
    }
}

File: app/src/main/java/com/rara/moviecatalog/MainActivity.kt
package com.rara.moviecatalog

import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.rara.moviecatalog.favorite.FavoriteFragment
import com.rara.moviecatalog.movie.MovieFragment
import com.rara.moviecatalog.setting.SettingsActivity
import com.rara.moviecatalog.tvshow.TvShowFragment
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    companion object {
        const val DATA_EXTRA = "data"
        const val MOVIE = "movie"
        const val TVSHOW = "tvShow"
        const val TYPE = "type"
        const val INSTANCE = "instance"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        nav_view.setOnNavigationItemReselectedListener { item ->
            when (item.itemId){
                R.id.navigation_movie -> {
                    val fragment = MovieFragment()
                    supportFragmentManager
                        .beginTransaction()
                        .replace(R.id.flContent, fragment, MovieFragment::class.java.simpleName)
                        .commit()
                    supportActionBar?.setTitle(R.string.title_movie)
                }
                R.id.navigation_tvshow -> {
                    val fragment = TvShowFragment ()
                    supportFragmentManager
                        .beginTransaction()
                        .replace(R.id.flContent, fragment, TvShowFragment::class.java.simpleName)
                        .commit()
                    supportActionBar?.setTitle(R.string.title_tv)
                }
                R.id.navigation_favorite -> {
                    val fragment = FavoriteFragment()
                    supportFragmentManager.beginTransaction()
                        .replace(R.id.flContent, fragment, FavoriteFragment::class.java.simpleName)
                        .commit()

                    supportActionBar?.setTitle(R.string.favorite)
                }
            }
            true
        }
        if (savedInstanceState == null) {
            nav_view.selectedItemId = R.id.navigation_movie
        }else{
            when (savedInstanceState.getString(INSTANCE)) {
                MovieFragment::class.java.simpleName -> {
                    nav_view.selectedItemId = R.id.navigation_movie
                }
                TvShowFragment::class.java.simpleName -> {
                    nav_view.selectedItemId = R.id.navigation_tvshow
                }
            }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.language_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.setting -> {
//                val mIntent = Intent(Settings.ACTION_LOCALE_SETTINGS)
                val mIntent = Intent(this, SettingsActivity::class.java)
                startActivity(mIntent)
            }
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onBackPressed() {
        moveTaskToBack(true)
        super.onBackPressed()
    }
}
